This workarounds GCC bug PR 95115 introduced in GCC 10. This is similar to
what is done in the float version.

--- a/sysdeps/ieee754/dbl-64/e_asin.c
+++ b/sysdeps/ieee754/dbl-64/e_asin.c
@@ -637,11 +637,7 @@ __ieee754_acos(double x)
   else
   if (k>0x7ff00000 || (k == 0x7ff00000 && u.i[LOW_HALF] != 0)) return x + x;
   else {
-    u.i[HIGH_HALF]=0x7ff00000;
-    v.i[HIGH_HALF]=0x7ff00000;
-    u.i[LOW_HALF]=0;
-    v.i[LOW_HALF]=0;
-    return u.x/v.x;
+    return (x-x)/(x-x);
   }
 }
 #ifndef __ieee754_acos
